dynamiqs/qarrays/dense_qarray.py

```diff
@@ -61,7 +61,7 @@ def conj(self) -> QArray:
         data = self.data.conj()
         return self._replace(data=data)
 
-    def reshape(self, *shape: int) -> QArray:
+    def _reshape_unchecked(self, shape: tuple[int, ...]) -> QArray:
```

Comment:
```suggestion
    def _reshape_unchecked(self, *shape: int) -> QArray:
```
We should keep the same API as `reshape`.

Reply:
the goal was to remove the syntaxic sugar in order to deter even more using it

Reply:
It's already a private and undocumented function. If users still want to use it, let's not give them some batons dans les roues. Plus we should keep some uniformity IMO.

Reply:
I don't agree but I won't argue on this

Reply:
Thank you!